### PR TITLE
BigTable: improve read rows validation performance

### PR DESCRIPTION
**Pull request by zakons**
Improve performance of reading multiple rows by reducing the number of times values are retrieved from a chunk, as this is very expensive.  Instead, move values from a chunk into a cell data object and access data from here in order to do validation.  includes other performance improvements as well.  Measurement were done with cProfile to find hotspots for tuning.

### CODE REVIEW

**Comment by zakons** [comment]
@tseaver Any ideas how to fix this docstring error, which breacks the circleci build?

`Warning, treated as error:
/var/code/gcp/.nox/docs/lib/python3.6/site-packages/google/cloud/bigquery/dbapi/cursor.py:docstring of google.cloud.bigquery.dbapi.Cursor.execute::more than one target found for cross-reference 'Any': google.cloud.bigquery_datatransfer_v1.types.Any, google.cloud.dataproc_v1.types.Any, google.cloud.firestore_v1beta1.types.Any, google.cloud.language_v1beta2.types.Any, google.cloud.spanner_admin_database_v1.types.Any, google.cloud.spanner_admin_instance_v1.types.Any, google.cloud.speech_v1.types.Any, google.cloud.videointelligence_v1beta1.types.Any, google.cloud.videointelligence_v1beta2.types.Any, google.cloud.vision_v1.types.Any
nox > Command bash ./test_utils/scripts/update_docs.sh failed with exit code 2
nox > Session docs failed. :(`

**Comment by chemelnucfin** [comment]
@zakons You should be able to pull from the latest master for now.

**Comment by tseaver** [comment]
This kind of thing is why I object to the aliasing done in all our various `types` modules #4715.  I suspect that all those `Any` classes are really aliases of the same underling `google.protobuf.descriptor_pb2.Any`.

It likely isn't an actual issue with the PR itself, only with our bad practice.

**Comment by zakons** [comment]
Please update CLA - providing my consent to the commits on this PR.

**Comment by chemelnucfin** [comment]
@zakons Technically, I believe that it is @aneepct who needs to consent since you are the original submitter...

File: spanner/benchmark/bin/ycsb [hunk 1]
```diff
@@ -1,20 +0,0 @@
-#!/bin/bash
```
Comment (chemelnucfin) [inline spanner/benchmark/bin/ycsb]: Why is this being taken out?

File: bigtable/google/cloud/bigtable/row_data.py [hunk 2]
```diff
@@ -267,20 +275,28 @@ def state(self):
         """State machine state.
 
         :rtype: str
-        :returns:  name of state corresponding to current row / chunk
+        :returns:  name of state corresponding to currrent row / chunk
```
Comment (chemelnucfin) [inline bigtable/google/cloud/bigtable/row_data.py]: extra r, but we can fix this later.

**Comment by sduskis** [comment]
@zakons and @jonparrott can this be merged?

**Comment by jonparrott** [comment]
Sure, but looking at the diff it's just a tiny change to a shell script? Can you re-confirm this is correct @sduskis and @zakons?

**Comment by chemelnucfin** [comment]
It might have been superceded by another PR?

**Comment by zakons** [comment]
@jonparrott @sduskis This PR is a rewrite of validation on the data coming in chunks from a ReadRowsResponse with the goal of improving read performance.  There is a major PR that should supersede this PR: Integration of GAPIC #5178. Another PR #4882, Retry on yield_rows was also recently merged that affected the same area of code. I will check if there are any changes from these 2 PR's that impact this PR and get back on the merge go-ahead decision as soon as possible.

**Comment by zakons** [comment]
@jonparrott The GAPIC PR #5178 is now ready to merge and, once done, please merge this PR next.

**Comment by sduskis** [comment]
@zakons This PR only has a spanner change in it.  Do you have any idea what happened here?

**Comment by sduskis** [comment]
@zakons: Do you know what happened to this PR?

**Comment by tseaver** [comment]
@zakons The net change for this PR is zero lines inside Bigtable.  I'm closing for now -- please feel free to re-open (or maybe better, push a cleaned-up branch and open a new PR).

**Pull request closed without merging**